src/conversions/cart.cpp

```diff
@@ -160,16 +163,16 @@ static PJ_LPZ geodetic(PJ_XYZ cart, PJ *P) {
 #else
     const double y_theta = cart.z * P->a;
     const double x_theta = p * P->b;
-    const double norm = hypot(y_theta, x_theta);
-    const double c = norm == 0 ? 1 : x_theta / norm;
-    const double s = norm == 0 ? 0 : y_theta / norm;
+    const double invnorm = invhypot(y_theta, x_theta);
```

Comment:
As you have seen invhypot is simply doing 1.0/sqrt(x*x + y*y) this is all in double precision. Ok, glibc seems to be claiming better than 2ulp v. 1ulp for hypot(). Will investigate. Except ctest, are there other special cases I should look at?
Bill

Reply:
As an additional sanity check I have rerun the latest ctest (master test/ Mar  1 17:10),
with g++ (GCC) 10.2.1. As expected, the master and the pull request behave the same. Ie both pass all ctest (except test 2 - geodesic-signtest, were, as noted in the comments,  glibc's cos(810) gives 6.12323e-17, rather than 0, due to rounding error in remquo).

Reply:
> I see that cart.cpp was mostly the work of @busstoptaktik .

While that is formally correct wrt. to the initial crude work, you @rouault did all the smart stuff speeding up the computations afterwards, so wrt. the statement

> I let it comment about the appropriateness of this change

I think you'll have more informed opinions on the speed and accuracy side of things.

Reply:
... however, I have a number of comments on the subject over in the Rust Geodesy version of [cart](https://github.com/busstoptaktik/geodesy/blob/main/src/ellipsoid/cartesians.rs).

The paper by Fukushima quoted there (full reference [here](https://github.com/busstoptaktik/geodesy/blob/main/src/bibliography/mod.rs) stubbornly insists on using `sqrt(x*x+y*y)`, and I think I have seen timings elsewhere claiming it to be 30 times faster than `hypot(x, y)`, which on the other hand is ½ulp more accurate...

Reply:
> While that is formally correct wrt. to the initial crude work, you @rouault did all the smart stuff speeding up the computations afterwards,

doh, I had already almost forgotten about that. You must refer to https://github.com/OSGeo/PROJ/commit/78c1df51e0621a4e0b2314f3af9478627e018db3

@wblangdon Could you try the bench code used in that commit? Or maybe the "./bin/bench_proj_trans" that is built with "make bench_proj_trans" does the job too.
We have a test failure on Linux 32 bit: https://github.com/OSGeo/PROJ/actions/runs/8157690552/job/22299140174?pr=4067

Reply:
sure:-)
I will be out of the office most of tomorrow (wednesday)
The system here defaults to 64bit, should be possible to switch it down to 32bit.
Bill

Reply:
Out of curiosity, I checked what the compiler is doing, and it was not helping me so much:
https://godbolt.org/z/en4Tend7x
Probably a `jmp` to `hypot` takes longer than a bunch of operations, including `sqrtsd`.

Have you tested the performance in in Windows?
https://godbolt.org/z/eKn7n6P3v

Reply:
> Probably a `jmp` to `hypot` takes longer than a bunch of operations, including `sqrtsd`.

@jjimenezshaw yes, because in general `hypot` implementations [are littered with defensive coding](https://opensource.apple.com/source/Libm/Libm-47.1/ppc.subproj/hypot.c.auto.html) to squeeze any additional half ulp of accuracy out in the result. `hypot` is for accuracy, not for speed

Reply:
As suggested I have tried 78c1df5,
https://github.com/OSGeo/PROJ/commit/78c1df51e0621a4e0b2314f3af9478627e018db3

I have run 78c1df5 as a benchmark (-O3 -DNDEBUG GCC 10.2.1, i7-4790 CPU @ 3.60GHz) 
with "fwd" the pull request is much the same (total, fwd+inv 9%) but with inv
alone the pull request is on average 17% faster.

I am away until Monday/Tuesday, plan next to investigate 32 bit.
Bill